packages/@react-aria/avatar/src/useAvatar.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaAvatarProps} from '@react-types/avatar';
+import {filterDOMProps, mergeProps} from '@react-aria/utils';
+import {ImgHTMLAttributes} from 'react';
+
+interface AriaAvatarOptions extends AriaAvatarProps {
+  /**
+   * The HTML element used to render the avatar, e.g. 'img', or 'div'.
+   *
+   * @default 'img'
+   */
+   elementType?: string
+}
+
+interface AvatarAria {
+  /**
+   * Props for the avatar element.
+   */
+  avatarProps: ImgHTMLAttributes<HTMLElement>
+}
+
+/**
+ * Provides the behavior and accessibility implementation for an avatar
+ * component. An avatar is a thumbnail representation of an entity, such as a
+ * user or an organization.
+ */
+export function useAvatar(props: AriaAvatarOptions): AvatarAria {
+  const {
+    elementType = 'img',
+    alt = '',
+    isDisabled,
+    ...otherProps
+  } = props;
+
+  const avatarProps = {
+    ...(elementType === 'img' && {alt}),
```

Comment:
I think if it's not labelled, then it's supposed to be aria-role="presentation"
@majornista ?
This is the dom in the default story
<img width="1361" alt="Screen Shot 2021-07-31 at 1 35 48 PM" src="https://user-images.githubusercontent.com/698229/127751906-8478667a-be19-4eb2-83b4-5939bcd2cd40.png">

Reply:
As far as screen readers are concerned, having `alt=""` is effectively the same as `role="presentation"`, but the latter is not widely supported:
> Screen readers also allow the use of WAI-ARIA to hide elements by using `role="presentation"`. However, currently, this feature is not as widely supported as using a null `alt` attribute.

Source: https://www.w3.org/WAI/tutorials/images/decorative/#image-used-as-part-of-page-design.

That said, and according to my understanding from the below [excerpt](https://www.w3.org/TR/wai-aria-1.1/#presentation) from WAI-ARIA, we'd want to add a `role="presentation"` if `img` is part of a containing element that provides the role and alternative text of the overall construct:
> Authors _**SHOULD NOT**_ provide meaningful alternative text (for example, use `alt=""` in HTML) when the `presentation` role is applied to an image.
> 
> In the following code sample, the containing `img` and is appropriately labeled by the caption paragraph. In this example the `img` element can be marked as presentation because the role and the text alternatives are provided by the containing element.
> 
> EXAMPLE 10
>  ```html
>  <div role="img" aria-labelledby="caption">
>    <img src="example.png" role="presentation" alt="">
>    <p id="caption">A visible text caption labeling the image.</p>
>  </div>
>  ```

Another thing is, since the `presentation` role causes a given element to be treated as having no role or to be removed from the accessibility tree, do want that for Avatar? It doesn't seem to me that we'd want to omit it from the accessibility tree. I personally see Avatar as being a functional/informative image vs being a decorative image, so if anything, having a prominent role for screen readers is more fitting IMO.

Reply:
fascinating, thanks for going the extra mile in your research on this one